subsys/net/lib/lwm2m/lwm2m_rw_senml_cbor.c

```diff
@@ -0,0 +1,901 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define LOG_MODULE_NAME net_lwm2m_senml_cbor
+#define LOG_LEVEL CONFIG_LWM2M_LOG_LEVEL
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include <stdio.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <inttypes.h>
+#include <ctype.h>
+#include <sys/util.h>
+
+#include <zcbor_common.h>
+#include <zcbor_decode.h>
+#include <zcbor_encode.h>
+
+#include "lwm2m_engine.h"
+#include "lwm2m_object.h"
+#include "lwm2m_rw_senml_cbor.h"
+#include "lwm2m_senml_cbor_decode.h"
+#include "lwm2m_senml_cbor_encode.h"
+#include "lwm2m_senml_cbor_types.h"
+#include "lwm2m_util.h"
+
+struct cbor_out_formatter_data {
+	/* Data */
+	struct lwm2m_senml input;
+
+	/* Storage for basenames and names ~ sizeof("/65535/999/") */
+	struct {
+		char names[CONFIG_LWM2M_RW_SENML_CBOR_RECORDS][sizeof("/65535/999/")];
+		size_t name_sz; /* Name buff size */
+		uint8_t name_cnt;
+	};
+};
+
+struct cbor_in_fmt_data {
```

Comment:
Would be good to make the naming consistent - either use `formatter` or `fmt` in both cases, `cbor_in_` and `cbor_out_`.